### PR TITLE
Adds in textured decals.

### PR DESCRIPTION
**Pull request by CRITAWAKETS**
worked on this while mapping a thing and regretting not having it

## About The Pull Request

Adds in turf decals which allow texturing tiles in specific areas instead of having it be the whole tile.

## Why It's Good For The Game

I think the image shows it off better than i can say, really.
## Changelog
:cl:
imageadd: Added in turf decals of textured tiles. You can now have them be diagonal!
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
This doesn't look like it's being used anywhere.
It'd be a good idea to actually implement the things you add to avoid code and resource bloat.

**Comment by CRITAWAKETS** [comment]
I'd say its a better idea to implement it first so that people could make implementations and if any of those would conflict, the best implementation would be added.
TLDR: put it in first so more people can actually do things with it

**Comment by Rohesie** [comment]
Not how we generally go with things.
Find a use for them, or add them when there's a use.

**Comment by CRITAWAKETS** [comment]
> Not how we generally go with things.
> Find a use for them, or add them when there's a use.

Except this is infact exactly what happened with textured tiles. It got merged, and people later did things with it.

Also how tile sidings went. We may not have these in maps now, but they'll be in maps later.

**Comment by Rohesie** [comment]
In general we avoid adding things without implementation unless we're absolutely sure they won't end up as unused bloat.
In this case, I don't know whether this will be used or not, so I assume the maptainer will. I'll poke them about it.

**Comment by Maurukas** [comment]
I'm looking at this and struggling to understand from the description, screenshots and diffs, what the intended use case for this is, or what effect it is supposed to achieve that cannot be done with current decals. 

Could you share the mapping where you used this, or wanted to use this?

**Comment by Mickyan** [comment]
> Also how tile sidings went. They were in any maps when introduced if i renember correctly. We may not have these in maps now, but they'll be in maps later.

When I first added trims they went unused for months until I started mapping them in myself and people noticed. Sidings were actually mapped in meta's bar from the start. 

Just pick a location to showcase some use cases, you don't have to do an entire map

**Comment by CRITAWAKETS** [comment]
here's an example using meta's boxing ring. While this isnt in the PR, its an example of what you can do.

Keep in mind that normally, tiles have to be entirely textured or not textured at all. Decals allow choosing which parts of the tile are textured and which parts arent.

**Comment by CRITAWAKETS** [comment]
Actual map example, you can see usage of the textured tiles just about everywhere.

btw, the reason why there are the tiles other than corner is simple
long variants on diagonals: used for two-length or higher tiles, decided to make it like that so that maps with a large amount of these could avoid the kilo hell known as 2 billion turf decals and i figured performance (and map maintainability, actually) mattered more than a tiny bit of storage space.
intersections: allows making "corner" size pieces. Also useful for shenanigans involving shifted pixels.
middle: mostly for shifted pixel shenanigans, could be used for whole tiles even if there's the "whole" decal.
whole: useful for when you want a textured whole tile where the pixels are darker or lighter than usual.

**Comment by Twaticus** [comment]
gonna pass on this
